### PR TITLE
Upat bounty

### PR DESCRIPTION
**Pull request by reddyn12**
Had to switch ```__name__``` with ```name``` because frozen dataclass overrides it with class name

### CODE REVIEW

**Comment by reddyn12** [comment]
This branch timed: ```PYTHONPATH=. python3 test/test_pattern_matcher.py  0.84s user 0.31s system 251% cpu 0.459 total```

https://github.com/reddyn12/tinygrad/tree/upat_speed timed: ```PYTHONPATH=. python3 test/test_pattern_matcher.py  0.80s user 0.27s system 187% cpu 0.569 total```

Going to write tests in dict form, commit, then refactor for ease of testing speed

File: tinygrad/codegen/uops.py [hunk 1]
```diff
@@ -134,102 +142,101 @@ def loop_collapse(loop_start, loop_end, compval, idx, mval, multconst):
 # this is symbolic 2.0
 constant_folder = PatternMatcher([
   # arange loop folding (early)
-  ({"uop": UOps.ALU, "arg": TernaryOps.WHERE, "vin": ({"uop": UOps.ALU, "arg": BinaryOps.CMPLT, "vin": (
-    {"uop": UOps.ALU, "arg": BinaryOps.ADD, "vin":
-      [{"__name__": "idx"}, {"uop": UOps.ALU, "arg": BinaryOps.MUL,
-        "vin": [{"__name__": "mval", "uop": UOps.CONST}, {"uop": UOps.RANGE, "vin": ({"__name__": "loop_start"}, {"__name__": "loop_end"})}]}]},
-      {"__name__": "compval", "uop": UOps.CONST})}, {"__name__": "multconst", "uop": UOps.CONST}, {"uop": UOps.CONST, "arg": 0})}, loop_collapse),
+  (UPat(uop = UOps.ALU, arg = TernaryOps.WHERE, vin = (UPat(uop = UOps.ALU, arg = BinaryOps.CMPLT, vin = (
```
[the remark this answers is not in the document]
Comment (reddyn12) [reply tinygrad/codegen/uops.py]: Converted to positional args

**Comment by reddyn12** [comment]
When #4744 gets merged, I'll convert tests to match UPat spec

**Comment by geohot** [comment]
So this is unreviewable by reading. How have you tested that that's 0 functional changes?

**Comment by reddyn12** [comment]
I realized testing pre UPat and post would be inconvenient if I'm writing the tests in UPat form. That's why I added dict form tests in #4744. Was the closing comment for that meant for this pr?

**Comment by reddyn12** [comment]
> So this is unreviewable by reading. How have you tested that that's 0 functional changes?

I tested against https://github.com/reddyn12/tinygrad/blob/upat_speed/test/test_pattern_matcher.py
Unable to get it merged to master, but they pass.

**Comment by geohot** [comment]
That's not a complete test. That's a subset. You need something like process replay to really test this.

**Comment by geohot** [comment]
Closed in favor of #4791, I trust that test plan

**Pull request closed without merging**